crates/client/src/client.rs

```diff
@@ -1080,6 +1102,32 @@ impl FuelClient {
         Ok(balance.amount)
     }
 
+    pub async fn balance_with_required_block_header(
+        &self,
+        owner: &Address,
+        asset_id: Option<&AssetId>,
+        required_block_height: u32,
+    ) -> io::Result<u128> {
+        let owner: schema::Address = (*owner).into();
+        let asset_id: schema::AssetId = match asset_id {
+            Some(asset_id) => (*asset_id).into(),
+            None => schema::AssetId::default(),
+        };
+        let query = schema::balance::BalanceQuery::build(BalanceArgs { owner, asset_id });
+        let balance: types::Balance = self
+            .query_with_headers(
+                query,
+                vec![(
+                    "REQUIRED_FUEL_BLOCK_HEIGHT".to_string(),
```

Comment:
Hmm, I'm not 100% satisfied with this name. Especially since we provide a block height as input to some queries. When squinting your eyes it could be confusing if the "required_bock_height" is part of the query, when it is more a notion of the state of the node to ensure we're in sync.

How would `REQUIRED_CHAIN_TIP_HEIGHT` sound to you? Here it's clear that we're referring to the block height of the chain tip.

Reply:
I agree that we could have a better name (maybe `CHAIN_HEAD` instead of `CHAIN_TIP`) , but if we change it we have to notify other teams as well as we are deviating from the spec: see for example https://github.com/FuelLabs/fuels-ts/issues/3443